proxy.go

```diff
@@ -518,12 +518,30 @@ func (rp *reverseProxy) getScope(req *http.Request) (*scope, int, error) {
 		c = rp.clusters[u.toCluster]
 		cu = c.users[u.toUser]
 	}
+
+	isTransparent := false
+	if u == nil {
```

Comment:
This nil check is already applied in line 541, why don't we move this block of code after that line (together with `rp.lock.RUnlock()` )? 

Would you mind also extract this logic to the dedicated function? Otherwise `getScope` becomes too complex to read

Reply:
The checks in 523 and in 541 are different. 
The logic here is
```
if user_is_not_set:
    attempt_to_use_transparent_user
if user_is_not_set:   # even transparent logic did not work
    return error
```
To be honest I do not see obvious issues here, while I
* add a comment or a couple of ones
* Ok, I'll try to move this logic to a function

Reply:
@Garnek20  , I've introduced getTransparentUser and getUser.
Not sure that it makes sense, it seems to look too heavy.

Although it is not that bad as getTransparentUser together with original logic in getScope.